### PR TITLE
[GameBoard] Initializing an empty game board

### PR DESCRIPTION
**Pull request by pranathip**
# Summary
Initializing the empty game board at the start of a 2048 game

# Images/videos

### CODE REVIEW

**Comment by pranathip** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/12" target="_blank">Graphite</a>  👈
    * **PR #13** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/13" target="_blank">Graphite</a> 
      * **PR #14** <a href="https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/14" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/12?utm_source=stack-comment).

File: src/GameBoard.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+  }
+  return false;
+};
+
+export const isFull = (board) => {
+  return !hasValue(board, 0);
+};
+
+const getRandomPosition = () => {
```
Comment (pranathip) [inline src/GameBoard.js:23]: This is the method that gets a random position/square for the game to continue

File: src/GameBoard.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  board[row][col] = 2;
+  return board;
+};
+
+const compress = (board) => {
+  const newBoard = getEmptyBoard();
+  for (let i = 0; i < board.length; i++) {
+    let colIndex = 0;
+    for (let j = 0; j < board[i].length; j++) {
+      if (board[i][j] !== 0) {
```
Comment (pranathip) [inline src/GameBoard.js:48]: this is a comment

File: src/GameBoard.js (hunk 1 above)
Comment (pranasaurus-rex) [reply src/GameBoard.js:23]: this looks great!

File: src/GameBoard.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const colPosition = Math.floor(Math.random() * 4);
+  return [rowPosition, colPosition];
+};
+
+export const generateRandom = (board) => {
+  if (isFull(board)) {
+    return board;
+  }
+
+  let [row, col] = getRandomPosition();
```
Comment (pranasaurus-rex) [inline src/GameBoard.js:34]: Let's rename this variable, otherwise LGTM!

**Review by pranasaurus-rex — APPROVED** [review]
LGTM!

**Comment by pranathip** [comment]
@pranathip started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/12).

**Comment by pranathip** [comment]
@pranathip merged this pull request with [Graphite](https://app.graphite.dev/github/pr/pranathip/pranathi-2048-react-app/12).

**Pull request merged**